### PR TITLE
Create unit and integration test 

### PR DESCRIPTION
**Pull request by ryanmt95**
Unit Test
 - player
 - room

Integration Test
 - room_controller (POST /join_room & POST /create_room)

### CODE REVIEW

File: gibberish_server/controllers/room_controller.js [hunk 1]
```diff
@@ -44,9 +44,12 @@ class RoomController {
         const roomId = req.body.roomId
         roomModel.getRoomInfoAsObject(roomId)
             .then(room => {
-                if(room.state != roomModel.STATE.GAME_WAITING) res.status(400).send('Game has started')
-                const index = room.players.findIndex(player => player.playerName == nickname)
-                if(index !== -1) {
+                console.log(room.players)
+                if (room.state != roomModel.STATE.GAME_WAITING) res.status(400).send('Game has started')
+                const index = room.players.findIndex(player => player.toJSON().playerName == nickname)
```
Comment (CiciliaHelena) [inline gibberish_server/controllers/room_controller.js:49]: why need to convert to json? can use player.name directly

File: gibberish_server/test/controller_test/room_test.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    assert.equal(res.status, 400)
+                    assert.equal(res.text, 'Game has started')
+                })
+                .catch(function (err) {
+                    console.log(err)
+                })
+        })
+    })
+
+})
```
Comment (CiciliaHelena) [inline gibberish_server/test/controller_test/room_test.js:104]: new line

File: gibberish_server/controllers/room_controller.js [hunk 3]
```diff
@@ -44,9 +44,12 @@ class RoomController {
         const roomId = req.body.roomId
         roomModel.getRoomInfoAsObject(roomId)
             .then(room => {
-                if(room.state != roomModel.STATE.GAME_WAITING) res.status(400).send('Game has started')
-                const index = room.players.findIndex(player => player.playerName == nickname)
-                if(index !== -1) {
+                console.log(room.players)
```
Comment (CiciliaHelena) [inline gibberish_server/controllers/room_controller.js:47]: unnecessary log

Comment (ryanmt95) [reply gibberish_server/controllers/room_controller.js:47]: yeap ok

File: gibberish_server/controllers/room_controller.js (hunk 1 above)
Comment (ryanmt95) [reply gibberish_server/controllers/room_controller.js:49]: hmmm yeah i'm not sure, the POST /join_room test (the one to prevent duplicate nicknames)kept failing if i didn't put that, let me try again..

File: gibberish_server/test/controller_test/room_test.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
+var chai = require('chai')
+    , chaiHttp = require('chai-http');
```
Comment (CiciliaHelena) [inline gibberish_server/test/controller_test/room_test.js:2]: var chaiHttp = require('chai-http');

File: gibberish_server/controllers/room_controller.js (hunk 1 above)
Comment (ryanmt95) [reply gibberish_server/controllers/room_controller.js:48]: yeah the test fails without the toJSON(), i'm not very sure why..

Comment (CiciliaHelena) [reply gibberish_server/controllers/room_controller.js:48]: i pass everything LOL
